### PR TITLE
WFLY-9720 Test calls QueryName.displayable() and checks amount of CPU it consumes.

### PR DESCRIPTION
**Pull request by dcihak**
Test for QueryName.displayable() added. Feel free to mention any ideas and opinions if this test is sufficient.

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/management/QueryNameDisplayableTestCase.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2012, Red Hat, Inc., and individual contributors
```
Comment (rsvoboda) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/management/QueryNameDisplayableTestCase.java]: It's 2018, not 2012 ;)

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/management/QueryNameDisplayableTestCase.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        String test = "select * from employee e (where e.id <> 0 and e.name ='foo' and e.other != 'other'";
+        long start = System.currentTimeMillis();
+
+        for ( int looper = 0; looper < 500000; looper ++) {
+            QueryName name = new QueryName(test);
+        }
+        long end = System.currentTimeMillis();
+        long duration = (end - start);
+
+        Assert.assertTrue("Duration of QueryName.displayable called in a 500000 loop must be lower than 3000 milliseconds, but was " + duration, duration < 3000);
```
Comment (rsvoboda) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/management/QueryNameDisplayableTestCase.java]: Do not use hardcoded limits / timeouts ... 

Try something like https://github.com/wildfly/wildfly-core/blob/master/testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/core/DeploymentOperationsTestCase.java#L113

Maybe @pkremens or @marekkopecky can point you to similar code in WF TS.

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/management/QueryNameDisplayableTestCase.java (hunk 2 above)
Comment (rsvoboda) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/management/QueryNameDisplayableTestCase.java]: `git grep TimeoutUtil` on wildfly repo shows quite many usages, feel free to get inspired there too.

Comment (pkremens) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/management/QueryNameDisplayableTestCase.java]: Confirmed, `TimeoutUtil` is the thing to be used here.

**Review by kabir — CHANGES_REQUESTED** [review]
@dcihak please make changes as per Rostislav's comments. Beyond that I have not looked at this at all

**Comment by dcihak** [comment]
@rsvoboda , @pkremens  Thanks a lot for the review and hints. I used the TimeoutUtil to compare with the duration.

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
@dcihak this test fails in the elytron test run (use -Delytron when running locally to reproduce). Actually it also fails on Windows too... Adding the fixme label mention `@wildfly/mergers-full` when done please

**Comment by stuartwdouglas** [comment]
Thanks for the PR, but I don't think we should be including tests like this in our test suite.

Our CI environment does not have consistent enough performance for this test to be meaningful (as you can see from the windows failure), and the performance limit is arbitrary.

**Pull request closed without merging**